testing/gcs-container/src/main/java/org/projectnessie/testing/gcs/GCSContainer.java

```diff
@@ -97,9 +100,16 @@ public GCSContainer() {
   public GCSContainer(String image) {
     super(image == null ? DEFAULT_IMAGE + ":" + DEFAULT_TAG : image);
 
+    ThreadLocalRandom rand = ThreadLocalRandom.current();
+    localAddress =
```

Comment:
I'm pretty sure this won't work on MacOS without fiddling with `ifconfig`, let me test that and get back to you.

Why not pick up a random available port and use it? Does the container absolutely requires 4443 and nothing else?

Reply:
Confirmed :-(

```
java.lang.RuntimeException: com.google.cloud.storage.StorageException: Connect timed out
	at org.projectnessie.testing.gcs.GCSContainer.start(GCSContainer.java:138)
	at org.projectnessie.gc.iceberg.inttest.ITSparkIcebergNessieGCP.setupGcs(ITSparkIcebergNessieGCP.java:38)
```

Can you at least add an `if(macOs) { ... }` branch?

Reply:
Ah, right. There was that behavior difference :()

Reply:
Fun fact: now I recall that I had a script for macOS in the past that added a ton of additional `127.x.y.z` addresses.